src/server/svr_jobfunc.c

```diff
@@ -839,13 +839,14 @@ svr_evaljobstate(job *pjob, int *newstate, int *newsub, int forceeval)
 			/* This is an array job. */
 			struct ajtrkhd  *ptbl = pjob->ji_ajtrk;
 			if (ptbl) {
-				if (ptbl->tkm_subjsct[JOB_STATE_QUEUED] >= ptbl->tkm_ct) {
+				if  (!(ptbl->tkm_subjsct[JOB_STATE_QUEUED] +
+				       ptbl->tkm_dsubjsct >= ptbl->tkm_ct)) {
```

Comment:
should we also consider sub jobs in 'W' state. 
Why is there a ">=", shouldn't "==" suffice? OR we can also use "<" and remove the "!".